src/components/AttachmentModal.js

```diff
@@ -115,7 +115,7 @@ class AttachmentModal extends PureComponent {
      * @param {String} sourceURL
      */
     downloadAttachment(sourceURL) {
-        fileDownload(sourceURL, this.props.originalFileName);
+        fileDownload(this.props.isAuthTokenRequired ? addEncryptedAuthTokenToURL(sourceURL) : sourceURL, this.props.originalFileName);
```

Comment:
For some reason this line is confusing me - how are we NOT adding the encrypted auth token to the URL for native here? I just tested Android and saw that the image request has the encrypted auth token in the correct header, NOT in the URL - but this line makes me think we're putting the `encryptedAuthToken` in the URL **ALWAYS** - can someone explain why I'm confused? 😅

Reply:
This is purely for the file download - both web and native get the encrypted auth token added to the URL here if required for the attachment. Attachments could be a PDF or image or some other file so I wanted to make sure that still used the `encryptedAuthToken` param here so all files download correctly.

As for the actual image request the source URL (without any auth token query param) and the `this.props.isAuthTokenRequired` gets passed down to the `ImageView` so that it can correctly add the auth token in the correct way for web (as a query param) and native (as a header token).

I hope that clears it up a bit 😅

Reply:
> Not sure whether any backend handling would need to be updated

As far as I know, we only currently use auth Tokens to get chat attachments, and I'm pretty sure all of that is working - passing the auth token in the header as opposed to the URL. If there's other places we pass `encryptedAuthToken` in a URL, we will probably have to modify the backend for those cases

> making fileDownload work with headers is not as important as images, because it's used a lot less often
on the other hand the logic would be cleaner and allow automatic cache usage

Good call, I thinkkk the benefits are useful enough to try to implement that on the next PR 👍

Reply:
Thanks for the review @kidroca, just going through your comments now 🙌 Just to confirm have we decided here to just use the `encryptedAuthToken` query param for file download then and set this up to use headers in a subsequent PR?